Proofreading fixes

SergioGasquez opened the pull request:
Thanks @f-hollow for all the suggestions.

Also, I will address some of your comments here:

> What about non-ESP32 chips, such as ESP8684, etc?

AFAIK, [ESP8684 is part of the ESP32-C2 series](https://www.espressif.com/en/products/socs/esp32-c2)

SergioGasquez reviewed (commented):
Here I address some of your comments/suggestions

f-hollow commented on src/writing-your-own-application/generate-project/esp-template.md:
````diff
@@ -46,7 +46,7 @@ Before going further, let's see what these files are for.
-## `main.rs`
+### Understanding `main.rs`
 
 ```rust,ignore
 #![no_std]
````
I agree that explanations can be added as comments in the code snippet. Longer explanations like for `#![no_main]` can be provided as additional information outside the code snippet.

Should some comments be included directly in `src/main.rs`?

SergioGasquez replied:
Another option would be to add line numbers (manually, as I haven't found any way to add them properly) and then split the code in chunks and add the explanation of each block below. Here is how manual line numbers look like in `mdbook`:
<img width="781" alt="image" src="https://github.com/esp-rs/book/assets/12926049/e86bb1e3-abf9-4e8c-b1be-95f232478380">

f-hollow replied:
This is also a good option. Actually, line numbers can help in orienting around the code if you split it into small chunks with individual comments as I initially proposed.

Please choose the approach that you like.

SergioGasquez replied:
Just tried with the blocks and line number, let me know your opinion

f-hollow replied:
I think splitting the code into three code snippets made explanations more contextualized. I like it!

If you like the result, suggest keeping it.

f-hollow commented on src/writing-your-own-application/generate-project/index.md:
```diff
@@ -25,28 +25,23 @@ Both templates are based on [cargo-generate], a tool that allows you to create a
-Dev Containers use the [`idf-rust` container image], that was explained in the [Using Container section] of the installation chapter, and provide an environment ready to develop Rust applications for Espressif chips with no installation required. Dev Containers also have integration with [Wokwi simulator], to simulate the project, and allow flashing from the container using [web flash].
-
-For more details on Dev Containers, see [Dev Container] section of the template README.
+Dev Containers use the [`idf-rust` container image], that was explained in the [Using Container] section of the [Setting up a Development Environment] chapter. This image provides an environment ready to develop Rust applications for Espressif chips with no installation required. Dev Containers also have integration with [Wokwi simulator], to simulate the project, and allow flashing from the cont ...
```
The section title is _Dev Containers_. Please fix that.

f-hollow commented on src/tooling/espflash.md:
```diff
@@ -1,20 +1,22 @@
+
 ## cargo-espflash
 
 Provides a subcommand for `cargo` that handles cross-compilation and flashing.
```
_Provides_ a subcommand or _is_ a subcommand?

f-hollow commented:
@SergioGasquez 

> We only cover VsCode as it's the most used IDE. In the past we had subsections for other IDEs, but they were very short and didn't provide much, so we decided to get rid of them.

Suggest briefly mentioning this:

 We chose VS Code because it has better support for Rust and is more popular. However, you can use other IDEs....

f-hollow commented:
There is also one comment about the Tips and Tricks section in `/src/tooling/text-editors-and-ides.md`

Looks like this section contains two tips:

- Issue with `std` and Rust Analyzer
- Custom toolchain and Cargo hints

Suggest splitting "Tips and Tricks" into two sections. As these sections relate to VS Code, suggest turning them into subsections VS Code.

f-hollow commented:
The only comment that we did not cover is about the word _ecosystem_ in "Ecosystem Overview."

The Ecosystem Overview chapter seems to cover the `std` and `no_std` development approaches. Can it be called an ecosystem?

f-hollow reviewed (commented):
Thank you for implementing my suggestions. Now only a few small things left.

f-hollow commented on src/overview/using-the-standard-library.md:
```diff
@@ -1,8 +1,6 @@
-Espressif provides a C-based development framework called [esp-idf][esp-idf-github] which has, or will have, support for all Espressif chips starting with the ESP32; note that this framework does _not_ support the ESP8266. See also [Support for Espressif Products][rust-esp-book-overview-index].
-
-[rust-esp-book-overview-index]: index.md#support-for-espressif-products
+Espressif provides a C-based development framework called [esp-idf][esp-idf-github] which has, or will have, support for all Espressif chips starting with the ESP32; note that this framework does _not_ support the ESP8266.
```
From what I understand, esp-idf has nothing to do with `no_std` development. The esp-idf support is only relevant from the perspective of `std`. So this should be broken down into two statements in the book:
 
* esp-idf is only relevant for `std` development
* `std` supports all the chips that esp-idf supports

SergioGasquez replied:
> From what I understand, esp-idf has nothing to do with no_std development.

Correct.

>  The esp-idf support is only relevant from the perspective of std. So this should be broken down into two statements in the book:
> 
> - esp-idf is only relevant for std development
> - std supports all the chips that esp-idf supports

Right, I think we are already sort of doing this. Do you have any suggestions?

f-hollow replied:
I think what we have now is good. I just proposed the way how to think of it to dispel any doubts.

f-hollow commented on src/overview/index.md:
```diff
@@ -8,7 +8,7 @@ There are the following approaches to using Rust on Espressif chips:
 
 - [Using the Standard Library (`std`)][rust-esp-book-std]
-- [Developing on Bare Metal (`no_std`)][rust-esp-book-no-std]
+- [Using the Core Library (`no_std`)][rust-esp-book-no-std]
```
Please also update the section heading.

f-hollow commented on src/tooling/debugging/vscode-debugging.md:
```diff
@@ -59,15 +58,22 @@ ESP32 doesn't have a built-in JTAG interface, so you have to connect an external
 
-ESP32-C3 with **revision 3** **does** have a built-in JTAG interface, and you don't have to connect an external device to be able to debug. To get the chip revision, run the `cargo espflash board-info` command.
+- Revisions older than 3 **don't** have built-in JTAG interface.
+- Revitions 3 (and newer) **do** have a built-in JTAG interface, and you don't have to connect an external device to be able to debug.
```
If revision 3 is the latest, it might be better to remove `(and newer)`.

SergioGasquez replied:
Not sure which revision is the latest, but I think there will be new revisions of it, so it might be worth it to leave it.

f-hollow replied:
Looks like there is an empty bullet point following this like. Please delete it.

f-hollow commented on src/tooling/espflash.md:
```diff
@@ -1,20 +1,22 @@
+
 ## cargo-espflash
 
 Provides a subcommand for `cargo` that handles cross-compilation and flashing.
```
_Provides_ a subcommand or _is_ a subcommand?

SergioGasquez replied:
In this case, `espflash` is a subcommand for `cargo`, so I think we are right saying that it provide a subcommand.

SergioGasquez commented:
> Suggest briefly mentioning this:
> 
> We chose VS Code because it has better support for Rust and is more popular. However, you can use other IDEs....

> Suggest splitting "Tips and Tricks" into two sections. As these sections relate to VS Code, suggest turning them into subsections VS Code.

Both things should be fixed in https://github.com/esp-rs/book/pull/109/commits/2cf23e4f5c1fa36976278b8906f3734ac49e04bc

SergioGasquez commented:
> The Ecosystem Overview chapter seems to cover the std and no_std development approaches. Can it be called an ecosystem? How about "Overview of Development Approaches" or something else?

Updated!

f-hollow commented:
@SergioGasquez 

The following steps might be (in this or separate PRs):

1. (Optional) Ask others to review the current updates.
2. Lint the content using Vale linter.
    - The configuration files are in [`idf/esp-vale-config`](https://gitlab.espressif.cn:6688/idf/esp-vale-config/)
    - As I haven't written the instructions yet, suggest having a call to discuss how to use Vale
3. Update file names and folder names in `src/`
    - Some names are outdated, so it is not easy to navigate inside `src/`
    - Consider also adding chapter numbers in file/folder names to arrange items in the order the book uses them

SergioGasquez commented:
Just pushed https://github.com/esp-rs/book/pull/109/commits/1b9e9bef94c8d58e7534fd05868463f29f0f1f01 updating  file names and folder names in `src/`and adding some redirections from old paths to new paths, so we maintain backward compatibility and do not break links pointing to the book. We already did break some links in the past, see https://github.com/esp-rs/esp-idf-template/pull/137 for example. 

With these redirections https://github.com/esp-rs/book/issues/72 can be also closed when this is merged. 

Adding numbers to the folder/file names will require adding a lot of redirections, and if we include/remove a chapter in the future, it will include a lot more redirections, for this reason I would prefer not to include them.

f-hollow commented:
@SergioGasquez 

Did another review and came up with a few more nitpicks. This is the last batch from me.

I uploaded a file with comments in [Code Review](https://marketplace.visualstudio.com/items?itemName=d-koppenhagen.vscode-code-review#:~:text=Simply%20right%20click%20somewhere%20in,the%20file%20and%20line%20references.) format on mattermost.

The book looks great!

SergioGasquez commented:
> @SergioGasquez
> 
> Did another review and came up with a few more nitpicks. This is the last batch from me.
> 
> I uploaded a file with comments in [Code Review](https://marketplace.visualstudio.com/items?itemName=d-koppenhagen.vscode-code-review#:~:text=Simply%20right%20click%20somewhere%20in,the%20file%20and%20line%20references.) format on mattermost.
> 
> The book looks great!
> 
> P.S. Just discovered that I can also attach files here. It is even more convenient.
> 
> [code-review.csv](https://github.com/esp-rs/book/files/12190048/code-review.csv) [code-review.csv.md](https://github.com/esp-rs/book/files/12190049/code-review.csv.md)

Just addressed all your comments/suggestions

f-hollow commented:
@SergioGasquez 

Thank you for implementing all the suggestions! One thing: You fixed the issue with the unnecessary scroll bar in the code snippet in _Understanding esp-template_. Please do the same in  _Understanding esp-idf-template_.

Now that you have formatted the notes and warnings according to the Rust Documentation Style Guide, the book looks as official as all other Rust books.

SergioGasquez commented:
> Thank you for implementing all the suggestions! One thing: You fixed the issue with the unnecessary scroll bar in the code snippet in _Understanding esp-template_. Please do the same in _Understanding esp-idf-template_.

Good catch! Fixed.
